src/util/pointer_offset_size.cpp

```diff
@@ -694,10 +694,16 @@ optionalt<exprt> get_subexpression_at_offset(
     if(*target_size_bits <= *elem_size_bits)
     {
       const mp_integer elem_size_bytes = *elem_size_bits / 8;
+      const auto offset_inside_elem = offset_bytes % elem_size_bytes;
+      const auto target_size_bytes = *target_size_bits / 8;
+      // only recurse if the cell completely contains the target
+      if(offset_inside_elem + target_size_bytes > elem_size_bytes)
+        return {};
```

Comment:
We could still use the `byte_extract_exprt` below I believe? So rather make the recursive call conditional?